agent/tool-scripts/postprocess/unittests

```diff
@@ -203,9 +195,18 @@ function testfio {
 
     mkdir ${_testroot}/$1
     cp -r samples/$1/* ${_testroot}/$1/
-
-    pbench_lib_dir=$(pwd)/../../lib pbench_bspp_dir=$(pwd)/../../bench-scripts/postprocess \
-        ../../bench-scripts/postprocess/fio-postprocess ${_testroot}/$1/result/reference-result fio- default > ${_testroot}/$1/stdout 2> ${_testroot}/$1/stderr
+    result_dir=${_testroot}/$1/result/reference-result
+
+    pbench_lib_dir=$(pwd)/../../lib \
+        pbench_bspp_dir=$(pwd)/../../bench-scripts/postprocess \
+        ../../bench-scripts/postprocess/fio-postprocess ${result_dir} fio- default \
+        > ${_testroot}/$1/stdout 2> ${_testroot}/$1/stderr
+
+    # Truncate the output file to at most $MAX_RESULT_LINES.  (There's a big fio
+    # test which produces literally millions of lines of output, and diff'ing
+    # them is painful.)
```

Comment:
I'd say the risk is substantial but that the downside of the false positive is better than the cost of avoiding it.

I don't recall what it "costs" to run `diff` on this particular file, but I have some experience with using `diff` on large JSON files, and it is, well, painful.  We would be better off with a specialized JSON comparison tool rather than `diff`, but, since I think we'd like to get rid of the gold file approach altogether, investing in something like that seems counterproductive (as adding compression/decompression support would be as well, I would think).

Reply:
What this test _should_ be doing is _timing_ its execution and failing if that exceeds a particular threshold -- that is, we don't care (too much) whether this test produces correct results -- what we care about is that it produces them in finite (and reasonable) time.

That is, in addition to true unit tests and proper functional tests, we really should have some performance tests, and this should form the basis for the fio ones...but we don't have anything like that at the moment.

Reply:
I don't think this is a gold file test in the end.  What we might want to do is generate the JSON file, and then write a Python script to ingest it and count the expected dictionaries created.

If we are able to generate the expected number without any errors being returned, and within 10 minutes (or some other long value), declare success.